frb_codegen/src/generator/dart/ty_general_list.rs

```diff
@@ -8,17 +8,43 @@ type_dart_generator_struct!(TypeGeneralListGenerator, IrTypeGeneralList);
 impl TypeDartGeneratorTrait for TypeGeneralListGenerator<'_> {
     fn api2wire_body(&self) -> Acc<Option<String>> {
         // NOTE the memory strategy is same as PrimitiveList, see comments there.
+        let contains_opacity = self
+            .ir
+            .inner
+            .distinct_types(self.context.ir_file)
+            .iter()
+            .any(IrType::is_opaque);
         Acc {
-            io: Some(format!(
-                "final ans = inner.new_{}_{}(raw.length);
-                for (var i = 0; i < raw.length; ++i) {{
-                    _api_fill_to_wire_{}(raw[i], ans.ref.ptr[i]);
+            io: if contains_opacity {
+                Some(format!(
+                    "
+                final ans = inner.new_{0}_{1}(raw.length);
+                try {{
```

Comment:
Same as above: merge the with-try and without-try codes to avoid duplicating too much